### ISSUE TITLE
"Move tabs to a new window" allows to lose data

### ISSUE THREAD

**Issue by QuAzI** [mozilla/multi-account-containers#558]
Move many many tabs into different area (work, personal, shopping). Click "Move tabs to a new window". Close window at end of day - all tabs closed with window and I don't know how to restore it. In cases when I want to lose data after window closed I can use other copy of Firefox or private mode. It is unacceptable to lose tabs

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/MAC-566)

**Comment by jonathanKingston** [comment]
@QuAzI can you be clearer about exactly what buttons what steps you are doing? What data you are losing?

I'm not following sorry.

**Comment by QuAzI** [comment]
I have lost all pages which I collected. This functional looks like Tab group in different window but if window closed - all data will be lost and can't be restored.

**Comment by jonathanKingston** [comment]
If you close a window in Firefox without closing the whole browser the tabs are not backed up. Is this what you mean?

This isn't really specific to containers from what I can gather, just that tabs aren't remembered on window close.

To my knowledge there are lots of session tab extensions which will cover this use case. I tried looking for some and asking for you however couldn't find anything. Writing something like this probably would be fairly easy.

I'm not exactly sure if this should be part of this extension specifically. What do you think @groovecoder?

**Comment by groovecoder** [comment]
Agree - it sounds like you want FIrefox to [save tabs on exit](https://support.mozilla.org/en-US/questions/786128) and/or [restore tabs from last time](https://support.mozilla.org/en-US/kb/how-do-i-restore-my-tabs-last-time). Those are not specific to containers though.

**Comment by QuAzI** [comment]
> If you close a window in Firefox without closing the whole browser the tabs are not backed up. Is this what you mean?
Yes

In my configuration Firefox save tabs on exit but only if it is not "Containers Tab in new window".
If addition means separate jobs, then they have to remain separate and save data. Like "Tab groups" addition it must save workplace. In other cases there is no reason to use this addition because "Private window" available and more predictable for one time windows. There is no sense to split tabs to tab containers if you need any time to open all tabs again and again and set target container to any tab. It is tabs hell instead simplification. Right now I have 44 tabs only on notebook (but I have home PC and work PC) and if I will open 44 tabs on any host and must sort by hands between containers day by day I will go crazy.

**Comment by QuAzI** [comment]
> If you close a window in Firefox without closing the whole browser the
tabs are not backed up. Is this what you mean?
Yes

In my configuration Firefox save tabs on exit but only if it is not
"Containers Tab in new window".
If addition means separate jobs, then they have to remain separate and
save data. Like "Tab groups" addition it must save workplace. In other
cases there is no reason to use this addition because "Private window"
available and more predictable for one time windows. There is no sense
to split tabs to tab containers if you need any time to open all tabs
again and again and set target container to any tab. It is tabs hell
instead simplification. Right now I have 44 tabs only on notebook (but I
have home PC and work PC) and if I will open 44 tabs on any host and
must sort by hands between containers day by day I will go crazy.

**Comment by jonathanKingston** [comment]
> Right now I have 44 tabs only on notebook (but I have home PC and work PC) 

Container tabs don't sync currently, this is an experiment. There are numerous bugs for this however it isn't a small job.

> In my configuration Firefox save tabs on exit but only if it is not "Containers Tab in new window".

Yeah I meant closing windows themselves, as I understand it:

1. You have a window with lots of different container tabs
2. You click on a container and click "Move tabs to a new window"
3. You close this new window whilst keeping the first window open

**Issue**: In 3. you expect some way to restore this container later.

Am I grasping what your issue is?

**Comment by QuAzI** [comment]
Yes, you are grasping. After container detached to new window the user has no way to save container tabs or move it again to main window.

**Comment by jonathanKingston** [comment]
Couple more questions:

1. Quitting Firefox completely will keep all windows on restore is that a huge issue?
2. Why would you move a large chunk of containers outside of the window only to move them back again? (can you explain your flow and why this is important please)
3. Can you not hide the container in the new window? (This should allow them to be restored later)

Maybe explaining your flow in #428 may help us design for your use case if it is a popular one.

**Comment by QuAzI** [comment]
1. If Firefox closed via hot key (Ctrl+Q) or via menu "File - Exit" - all windows closed in one time and will be restored in next run. But menu hidden by default and many users don't remember hotkeys. Default behaviour: user close Firefox window by window. Main window closed last to prevent tabs lost.
2. I try to split one big place to different containers (as in example: "work", "banking", "shopping") and split one window do different work places. I didn't plan to return back I haven't learned yet about an opportunity to lose tabs. I can't move it back to main window after I learned about data loss posibility.
3. Yes I can but it reduces the importance of a containers because I can't split workplaces. I have many tabs and rainbow in a top of browser will only distract from work. "Tab groups" is more useful in this case (but this addition will lost after FF57)

**Comment by jonathanKingston** [comment]
1. This is the default behaviour of all of Firefox, it got me after coming back to Firefox from a long time being a Chrome user. This was my main push back. I think there should be a nice extension to fix this and ideally it might become something native in Firefox too.

2. Data loss is solved by whatever 1. does.

3. I'm unclear the advantage of the tab groups extension coupled with containers here. Writing example flows of how you use containers might help in #428. How you browse and create new windows, what they are for. Example screenshots might help too.

In short I still think the main issue is window closing doesn't permit session restoring. This seems unrelated to if you have containers in these windows. We have an option to snap a container into a new window however this is just an aid on other tab features. I think this is out of scope for our little experiment. I also think some of your issues are duplicates of the comment in #336.

I feel like there was a place for non developers to ask for such an extension, I will try and find this.

**Comment by aleksejrs** [comment]
> Why would you move a large chunk of containers outside of the window only to move them back again? (can you explain your flow and why this is important please)

There is no confirmation for that action, so you can move them by accident.

**Comment by grahamperrin** [comment]
(Following this with great interest …)

**Comment by grahamperrin** [comment]
#943
